### DISCUSSION TITLE
Is there any IPython-based documentation for direct implementation in Python programs?

### DISCUSSION THREAD

**Discussion by wyzzrt** [talmolab/sleap#2070] [General]
Hi,

Thank you for providing such a great tool. I’ve been trying to compare your method with DeepLabCut (DLC), but I noticed that there don’t seem to be Python-based implementations mentioned in your documentation. From what I’ve seen, it appears the functionality can only be accessed through the GUI or comand lines.

I’m wondering if there’s any plan to provide a IPython-based documentation for direct implementation in Python programs. Without such an interface, it feels like there might be some limitations or issues that are being hidden from users, as full flexibility is often crucial for research and custom analysis (Just like DeepLabCut).

Looking forward to your response. Thank you for your hard work!

**Comment by talmo** [comment]
Hi @wyzzrt,

Yep, we have a couple of notebooks in the docs for exactly this:

- [**Interactive and resumable training**](https://sleap.ai/notebooks/Interactive_and_resumable_training.html)
- [**Interactive and realtime inference**](https://sleap.ai/notebooks/Interactive_and_realtime_inference.html)

Then there's the [**API reference**](https://sleap.ai/api.html) and you might also be interested in [`sleap-io`](https://io.sleap.ai/latest/) for easier data manipulation.

Let us know if there's something you're looking for that's not covered in those!

Cheers,

Talmo

**Comment by roomrys** [comment]
Hello,

I am closing this for now, but please re-open if you need further guidance @wyzzrt!

Thanks,
Liezl
